[help patch] whatis vs. ptype - the difference #2
http://sourceware.org/ml/gdb-patches/2011-07/msg00771.html
http://sourceware.org/ml/gdb-cvs/2011-07/msg00227.html

### src/gdb/ChangeLog	2011/07/27 17:08:05	1.13232
### src/gdb/ChangeLog	2011/07/27 17:41:18	1.13233
## -1,5 +1,10 @@
 2011-07-27  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
+	* typeprint.c (_initialize_typeprint): Extend the help of "whatis" and
+	"ptype" by their typedefs difference.
+
+2011-07-27  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
 	* dwarf2expr.c (ctx_no_read_reg): New function.
 	* dwarf2expr.h (ctx_no_read_reg): New declaration.
 	* dwarf2read.c (read_2_signed_bytes, read_4_signed_bytes): Remove.
--- src/gdb/typeprint.c	2011/01/11 21:53:25	1.41
+++ src/gdb/typeprint.c	2011/07/27 17:41:19	1.42
@@ -316,8 +316,10 @@
 Print definition of type TYPE.\n\
 Argument may be a type name defined by typedef, or \"struct STRUCT-TAG\"\n\
 or \"class CLASS-NAME\" or \"union UNION-TAG\" or \"enum ENUM-TAG\".\n\
-The selected stack frame's lexical context is used to look up the name."));
+The selected stack frame's lexical context is used to look up the name.\n\
+Contrary to \"whatis\", \"ptype\" always unrolls any typedefs."));
 
   add_com ("whatis", class_vars, whatis_command,
-	   _("Print data type of expression EXP."));
+	   _("Print data type of expression EXP.\n\
+Only one level of typedefs is unrolled.  See also \"ptype\"."));
 }
### src/gdb/testsuite/ChangeLog	2011/07/27 17:08:05	1.2813
### src/gdb/testsuite/ChangeLog	2011/07/27 17:41:20	1.2814
## -1,5 +1,9 @@
 2011-07-27  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
+	* gdb.base/help.exp (help whatis): Update the expected string.
+
+2011-07-27  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
 	* gdb.dwarf2/dw2-simple-locdesc.S: New file.
 	* gdb.dwarf2/dw2-simple-locdesc.exp: New file.
 	* gdb.dwarf2/dw2-stack-boundary.exp (check partial symtab errors):
--- ./gdb/testsuite/gdb.base/help.exp.orig	2010-07-28 01:21:07.000000000 +0200
+++ ./gdb/testsuite/gdb.base/help.exp	2011-07-27 19:52:49.000000000 +0200
@@ -671,7 +671,7 @@ test_class_help "user-defined" {
 # test help watch
 gdb_test "help watch" "Set a watchpoint for an expression\.\[\r\n\]+A watchpoint stops execution of your program whenever the value of\[\r\n\]+an expression changes\." "help watch"
 # test help whatis
-gdb_test "help whatis" "Print data type of expression EXP\." "help whatis"
+gdb_test "help whatis" "Print data type of expression EXP\.\r\nOnly one level .*" "help whatis"
 # test help where
 gdb_test "help where" "Print backtrace of all stack frames, or innermost COUNT frames\.\[\r\n\]+With a negative argument, print outermost -COUNT frames\.\[\r\n\]+Use of the 'full' qualifier also prints the values of the local variables\." "help where"
 # test help x
